shaders/pbr-frag.glsl

```diff
@@ -53,257 +66,237 @@ varying vec3 v_Normal;
-// Implementation from “A Reflectance Model for Computer Graphics” by Robert Cook and Kenneth Torrance,
-float geometricOcclusionCookTorrance(PBRInfo pbrInputs)
+// Calculation of the lighting contribution from an optional Image Based Light source.
+// Precomputed Environment Maps are required uniform inputs and are computed as outlined in [1].
+// See our README.md on Environment Maps [3] for additional discussion.
+vec3 getIBLContribution(PBRInfo pbrInputs, vec3 n, vec3 reflection)
 {
-  return min(min(2.0 * pbrInputs.NdotV * pbrInputs.NdotH / pbrInputs.VdotH, 2.0 * pbrInputs.NdotL * pbrInputs.NdotH / pbrInputs.VdotH), 1.0);
-}
+    float mipCount = 9.0; // resolution of 512x512
+    float lod = (pbrInputs.perceptualRoughness * mipCount);
+    // retrieve a scale and bias to F0. See [1], Figure 3
+    vec3 brdf = texture2D(u_brdfLUT, vec2(pbrInputs.NdotV, 1.0 - pbrInputs.perceptualRoughness)).rgb;
+    vec3 diffuseLight = textureCube(u_DiffuseEnvSampler, n).rgb;
+
+#ifdef USE_TEX_LOD
+    vec3 specularLight = textureCubeLodEXT(u_SpecularEnvSampler, reflection, lod).rgb;
+#else
+    vec3 specularLight = textureCube(u_SpecularEnvSampler, reflection).rgb;
+#endif
 
-// implementation of microfacet occlusion from “An Inexpensive BRDF Model for Physically based Rendering” by Christophe Schlick
-float geometricOcclusionSchlick(PBRInfo pbrInputs)
-{
-  float k = pbrInputs.perceptualRoughness * 0.79788; // 0.79788 = sqrt(2.0/3.1415); perceptualRoughness = sqrt(alphaRoughness);
-  // alternately, k can be defined with
-  // float k = (pbrInputs.perceptualRoughness + 1) * (pbrInputs.perceptualRoughness + 1) / 8;
+    vec3 diffuse = diffuseLight * pbrInputs.diffuseColor;
+    vec3 specular = specularLight * (pbrInputs.specularColor * brdf.x + brdf.y);
 
-  float l = pbrInputs.LdotH / (pbrInputs.LdotH * (1.0 - k) + k);
-  float n = pbrInputs.NdotH / (pbrInputs.NdotH * (1.0 - k) + k);
-  return l * n;
-}
+    // For presentation, this allows us to disable IBL terms
+    diffuse *= u_ScaleIBLAmbient.x;
+    specular *= u_ScaleIBLAmbient.y;
 
-// the following Smith implementations are from “Geometrical Shadowing of a Random Rough Surface” by Bruce G. Smith
-float geometricOcclusionSmith(PBRInfo pbrInputs)
-{
-  float NdotL2 = pbrInputs.NdotL * pbrInputs.NdotL;
-  float NdotV2 = pbrInputs.NdotV * pbrInputs.NdotV;
-  float v = ( -1.0 + sqrt ( pbrInputs.alphaRoughness * (1.0 - NdotL2 ) / NdotL2 + 1.)) * 0.5;
-  float l = ( -1.0 + sqrt ( pbrInputs.alphaRoughness * (1.0 - NdotV2 ) / NdotV2 + 1.)) * 0.5;
-  return (1.0 / max((1.0 + v + l ), 0.000001));
+    return diffuse + specular;
 }
 
-float SmithG1_var2(float NdotV, float r)
+// Basic Lambertian diffuse
+// Implementation from Lambert's Photometria https://archive.org/details/lambertsphotome00lambgoog
+// See also [1], Equation 1
+vec3 diffuse(PBRInfo pbrInputs)
 {
-  float tanSquared = (1.0 - NdotV * NdotV) / max((NdotV * NdotV), 0.00001);
-  return 2.0 / (1.0 + sqrt(1.0 + r * r * tanSquared));
```

Comment:
`SmithG1_var2` was our choosen implementation for geometricOcclusion before this PR, since it was added back in https://github.com/KhronosGroup/glTF-WebGL-PBR/commit/507b8df2490bf60f62781e109ed54b50dedd3d7c

I switched us back to var1 (it is now just inlined in the `geometricOcclusion` function), as I could not find the above implementation in the literature, and saw var1 used rampantly.  Would like input from @snagy on this particular change, incase we are switching away from a more accurate/efficient occlusion implementation.